### PR TITLE
Update github actions

### PR DESCRIPTION
**Pull request by matriv**
## Summary of the changes / Why this improves CrateDB

- Update github action `checkout` to v3

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -35,12 +35,19 @@ jobs:
         with:
           arguments: :server:test -Dtests.crate.run-windows-incompatible=true
 
+      - name: Setup tmate session
+        if: ${{ failure() }}
+        uses: mxschmitt/action-tmate@v3
+        timeout-minutes: 15
+        with:
+          limit-access-to-actor: true
```
Comment (amotl) [inline .github/workflows/main.yml:43]: Thanks for adding this configuration setting!

**Comment by matriv** [comment]
Not insisting to enable tmate, I was mostly play around and thought it could be useful.
@mfussenegger Waiting for your opinion as well.

**Review by mfussenegger — COMMENTED** [review]
I'm not sure enabling tmate automatically on a failure is practical.

Given our workflow (first review, then add a label to merge) we're usually not actively monitoring the PRs and 15 minutes would be a tight window to connect.

On the other hand, anything longer would most of the time consume additional resources for nothing.

And it uses a service running on tmate.io, which from the looks of it a single person is paying from their own pockets.


What were you trying to look for when connecting with SSH? Can we instead expose more details in the logs?

**Comment by matriv** [comment]
I was mostly playing around with this option, with a simple thought that one could login and maybe rerun with the seed and maybe see something useful that helps consistently reproduce a failure.

I also agree that the options for a smooth workflow are not good enough, and the timeout is not helping much.

Didn't check the tmate.io to be honest, but what you say is another -1 to go ahead with is.

I'll drop the commit and only merge the upgrade of GitHub used actions to newer versions.

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**